### DISCUSSION TITLE
Duplication of method execution in two different threads.

### DISCUSSION THREAD

**Discussion by ElVargas95** [playframework/playframework#11939] [Q&A]
We are using Play Framework 2.8.20 for Java, and we are facing an issue with a method that we are consuming through an Ajax client. The problem we are encountering is that when executing the transaction, as it is a heavy transaction, the method reaches a point where it spawns another thread and starts its execution again. This leads to the method's execution being duplicated twice, which is a serious issue in production. What can we do to prevent the server from making these double attempts without notifying it?

**Comment by mkurz** [comment]
Sorry, I do not fully understand. So the method is an action method and you do send data TO an ajax client?
And that action method does _on purpose_ start another thread? So you implemented the logic to start another thread?
What do you mean ..."without notifying it?"
